### ISSUE TITLE
"Type casting can be used" inspection doesn't differentiate between casting and coercion

### ISSUE THREAD

**Issue by Bilge** [kalessil/phpinspectionsea#872]
The following type coercion code generates an inspection warning:

```php
$this->foo = "$foo";
```

It incorrectly claims:
>'(string) $foo' should be used instead (up to 6x faster)

Perhaps it is true that `(string)$foo` is faster than `strval($foo)`, which is what the inspection claims to be checking for, but it's absolutely incorrect that coercion (which it doesn't even claim to be checking for) is 6x slower than casting.

This inspection should stick to checking the PHP4 functions (intval, floatval, doubleval, strval) as advertised, and leave false claims about coercion out of it. Regardless of speed, it's a style preference and not strictly incorrect to use coercion over casting.

**Comment by kalessil** [comment]
Thank you for reporting @Bilge 

For myself: update message, introduce settings so people could switch off the case reporting.
